pkg/cluster/builder.go

```diff
@@ -55,7 +55,7 @@ func NewFileSpecBuilder(reader manifests.FileReader, cliVersion version.Info, op
 
 // Build constructs a new [Spec] by reading the cluster config in yaml from a file and
 // Releases, Bundles and EKS-D manifests from the configured URLs.
-func (b FileSpecBuilder) Build(clusterConfigURL string) (*Spec, error) {
+func (b FileSpecBuilder) Build(clusterConfigURL string, skipIPCheck bool) (*Spec, error) {
```

Comment:
we should avoid adding any logic here for "options"

can we have something a like a "defaulter" service?

Maybe it's time we start using this
https://github.com/aws/eks-anywhere/blob/main/designs/validations.md